Packs/PaloAltoNetworks_Threat_Vault/Integrations/Threat_Vault/Threat_Vault.py

```diff
@@ -0,0 +1,485 @@
+        self._params = {'api_key': api_key}
+        self.name = 'ThreatVault'
+
+    def antivirus_signature_get_request(self, sha256: str = '', signature_id: str = '') -> dict:
+        """Get antivirus signature by sending a GET request.
+
+        Args:
+            sha256: antivirus sha256.
+            signature_id: signature ID.
+        Returns:
+            Response from API.
+        """
+        if (sha256 and signature_id) or (not sha256 and not signature_id):
+            raise Exception('Please submit a sha256 or a signature_id.')
+        if signature_id:
+            suffix = f'/threatvault/panav/signature/{signature_id}'
+        else:
+            suffix = f'/file/{sha256}/signature'
+
+        return self._http_request(method='GET', url_suffix=suffix, params=self._params)
+
+    def dns_signature_get_request(self, dns_signature_id: str) -> dict:
+        """Get DNS signature by sending a GET request.
+
+        Args:
+            dns_signature_id: DNS signature ID.
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET', url_suffix=f'/threatvault/dns/signature/{dns_signature_id}',
+                                  params=self._params)
+
+    def antispyware_get_by_id_request(self, signature_id: str) -> dict:
+        """Get DNS signature by sending a GET request.
+
+        Args:
+            signature_id: signature ID.
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET', url_suffix=f'/threatvault/ips/signature/{signature_id}',
+                                  params=self._params)
+
+    def ip_geo_get_request(self, ip_: str) -> dict:
+        """Get IP geolocation by sending a GET request.
+
+        Args:
+            ip_: ip address.
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET', url_suffix=f'/ip/{ip_}/geolocation', params=self._params)
+
+    def search_request(self, path: str, from_: int, to_: int, signature_name: str = '', domain_name: str = '',
+                       vendor: str = '', cve: str = '') -> dict:
+        """Initiate a search by sending a POST request.
+
+        Args:
+            path: API endpoint path to search.
+            from_: from which signature to return results
+            to_: to which signature to return results
+            signature_name: signature name.
+            domain_name: domain name.
+            vendor: vendor ID.
+            cve: cve ID.
+        Returns:
+            Response from API.
+        """
+        if path == 'dns':  # DNS search
+            if signature_name and domain_name:
+                raise Exception('Please provide either a signature_name or a domain_name.')
+        elif path == 'ips':  # Anti spyware search
+            if (cve and (vendor or signature_name)) or (vendor and (cve and signature_name)) \
+                    or (signature_name and (cve or vendor)):
```

Comment:
```suggestion
            if (cve and (vendor or signature_name)) or (vendor and (cve or signature_name)) \
                    or (signature_name and (cve or vendor)):
```
maybe this is what you meant?
consider covering this in UT.